src/index.ts

```diff
@@ -202,7 +199,11 @@ export async function sign(payload: JwtPayload, secret: string, options: JwtSign
     const partialToken = `${base64UrlStringify(_utf8ToUint8Array(JSON.stringify({ ...options.header, alg: options.algorithm })))}.${base64UrlStringify(_utf8ToUint8Array(payloadAsJSON))}`
     let keyFormat = 'raw'
     let keyData
-    if (secret.startsWith('-----BEGIN')) {
+    if (typeof secret === 'object') {
```

Comment:
According to the type definitions `secret` can only be a `string`. This should be typed correctly.